### ISSUE TITLE
Incorrect duplicate test for \SplFileObject

### ISSUE THREAD

**Issue by cjtaylor-adepteo** [XeroAPI/xero-php-oauth2#280]
The \SplFileObject on line 311 blocks what I believe is the correct match on 314 and stops PDFs invoices being downloaded.

https://github.com/XeroAPI/xero-php-oauth2/blob/88e2d6bec113d0f0f4186ccd76180cf06bdec7e8/lib/AccountingObjectSerializer.php#L311

```
  311     ⦙   //} elseif (in_array($class, ['\DateTime', '\SplFileObject', 'array', 'bool', 'boolean', 'byte', 'double', 'float', 'int', 'integer', 'mixed', 'number', 'object', 'string', 'void'], true)) {           
  312     ⦙   } elseif (in_array($class, ['\DateTime',  'array', 'bool', 'boolean', 'byte', 'double', 'float', 'int', 'integer', 'mixed', 'number', 'object', 'string', 'void'], true)) {                                 
  313     ⦙   ⦙   settype($data, $class);                                                                                                                                                                                 
  314     ⦙   ⦙   return $data;                                                                                                                                                                                           
  315     ⦙   } elseif ($class === '\SplFileObject') {  
```

**Comment by eliissc** [comment]
We are facing the same issue: the SDK doesn't return an SplFileObject anymore and we are unable to download the attachments.
I did the same change directly in the SDK that the issue creator pointed out. I hope this will be fixed in the next version.

**Comment by mbardelmeijer** [comment]
See https://github.com/XeroAPI/xero-php-oauth2/pull/277 as well.

**Comment by RettBehrens** [comment]
looking to resolve with #281

**Comment by pumpkinball** [comment]
Fixed on latest release 2.14.0
